spt/features/strafehud.cpp

```diff
@@ -0,0 +1,533 @@
+
+	const __m128 accel = _mm_sub_ps(newVelLen, _mm_set1_ps(oldVelLength2D));
+
+	const __m128 result = _mm_blendv_ps(accel, _mm_set1_ps(0.0f), _mm_or_ps(zeroMask, accelMask));
+
+	_mm_storeu_ps(outAccel, result);
+}
+
+void StrafeHUD::SetData(uintptr_t pCmd)
+{
+	if (!spt_strafehud.GetBool())
+		return;
+
+	const CUserCmd* cmd = reinterpret_cast<CUserCmd*>(pCmd);
+	float forwardmove = cmd->forwardmove;
+	float sidemove = cmd->sidemove;
+	float upmove = cmd->upmove;
+
+	accels.clear();
+	lines.clear();
+
+	const auto player = spt_playerio.GetPlayerData();
+	const auto vars = spt_playerio.GetMovementVars();
+	const float playerYaw = utils::GetPlayerEyeAngles().y;
+
+	float relAng = 0.0f;
+	const int lockMode = spt_strafehud_lock_mode.GetInt();
+	if (lockMode > 0)
+	{
+		relAng = playerYaw;
+		if (lockMode == 1)
+		{
+			QAngle angles;
+			VectorAngles(player.Velocity, Vector(0, 0, 1), angles);
+			relAng -= angles.y;
+		}
+		relAng *= utils::M_DEG2RAD;
+	}
+
+	const float speed = forwardmove * forwardmove + sidemove * sidemove + upmove * upmove;
+	if (speed > vars.Maxspeed * vars.Maxspeed)
+	{
+		const float ratio = vars.Maxspeed / std::sqrtf(speed);
+		forwardmove *= ratio;
+		sidemove *= ratio;
+		upmove *= ratio;
+	}
+
+	wishDir = Vector(std::cosf(relAng) * sidemove - std::sinf(relAng) * forwardmove,
+	                 std::sinf(relAng) * sidemove + std::cosf(relAng) * forwardmove,
+	                 0.0f);
+	const float wishDirLen = wishDir.Length2D();
+	if (wishDirLen > 1.0f)
+	{
+		wishDir /= wishDirLen;
+	}
+
+	int detail = spt_strafehud_size.GetInt() * spt_strafehud_detail_scale.GetFloat();
+	detail = ((detail + 3) / 4) * 4; // round up to multiple of 4
+	accels.reserve(detail);
+
+	const Vector oldVel = GetGroundFrictionVelocity(player, vars);
+
+	const float sinPlayerYaw = std::sinf(playerYaw * utils::M_DEG2RAD);
+	const float cosPlayerYaw = std::cosf(playerYaw * utils::M_DEG2RAD);
+
+	for (int i = 0; i < detail; i += 4)
+	{
+		std::array<float, 4> ang;
+		for (int j = 0; j < 4; j++)
+		{
+			ang[j] = ((i + j) / (float)detail) * 2.0f * M_PI + relAng;
+		}
+
+		std::array<float, 4> accel{};
+		GetAccelAfterMove(player,
+		                  vars,
+		                  sinPlayerYaw,
+		                  cosPlayerYaw,
+		                  oldVel,
+		                  oldVel.Length2D(),
+		                  ang.data(),
+		                  accel.data());
+		accels.insert(accels.end(), accel.begin(), accel.end());
+	}
+
+	auto [minIt, maxIt] = std::minmax_element(accels.begin(), accels.end());
+	float biggestAccel = *maxIt;
+	float smallestAccel = *minIt;
+	for (float& accel : accels)
+	{
+		if (accel > 0.0f && biggestAccel > 0.0f)
+		{
+			accel /= biggestAccel;
+		}
+		else if (accel < 0.0f && smallestAccel < 0.0f)
+		{
+			accel /= -smallestAccel;
+		}
+	}
+
+	shouldUpdateLines = true;
+}
+
+void StrafeHUD::DrawHUD()
+{
+	interfaces::surface;
```

Comment:
this looks like a leftover from something

Reply:
Fixed :)
Also added strafehud options in spt_gui